hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/FlinkRowDataReaderContext.java

```diff
@@ -120,6 +124,9 @@ public Option<HoodieRecordMerger> getRecordMerger(RecordMergeMode mergeMode, Str
       case COMMIT_TIME_ORDERING:
         return Option.of(CommitTimeFlinkRecordMerger.INSTANCE);
       default:
+        if (mergeStrategyId.equals(HoodieRecordMerger.PAYLOAD_BASED_MERGE_STRATEGY_UUID)) {
```

Comment:
why needs special handling here? should `HoodieRecordUtils.createValidRecordMerger` cover this instead?